// @moralisweb3/auth/lib/Auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Auth = exports.BASE_URL = void 0;
var common_core_1 = require("@moralisweb3/common-core");
var requestMessage_1 = require("./methods/requestMessage");
var verify_1 = require("./methods/verify");
exports.BASE_URL = 'https://authapi.moralis.io';
var Auth = /** @class */ (function () {
    function Auth(core) {
        var _this = this;
        this.core = core;
        this.name = Auth.moduleName;
        this.type = common_core_1.ModuleType.API;
        this.baseUrl = exports.BASE_URL;
        this.requestMessage = function (options) { return (0, requestMessage_1.makeRequestMessage)(_this.core)(options); };
    }
    Auth.create = function (core) {
        return new Auth(core !== null && core !== void 0 ? core : common_core_1.CoreProvider.getDefault());
    };
    Auth.prototype.verify = function (options) {
        return (0, verify_1.makeVerify)(this.core)(options);
    };
    Auth.moduleName = 'auth';
    return Auth;
}());
exports.Auth = Auth;
